modules/references/server/controllers/reference.server.controller.js

```diff
@@ -0,0 +1,215 @@
+'use strict';
+
+var mongoose = require('mongoose'),
+    _ = require('lodash'),
+    path = require('path'),
+    async = require('async'),
+    errorService = require(path.resolve('./modules/core/server/services/error.server.service')),
+    emailService = require(path.resolve('./modules/core/server/services/email.server.service')),
+    pushService = require(path.resolve('./modules/core/server/services/push.server.service')),
+    Reference = mongoose.model('Reference'),
+    User = mongoose.model('User');
+
+/**
+ * Validate the request body and data consistency
+ * of Create a reference
+ */
+function validateCreate(req) {
```

Comment:
the request validation could use some abstraction or some library.
_disclaimer:_ i'm not fully aware what _abstraction_ means. i just have _ideas_ about it. :)

Reply:
#807